lib/config/applyrule.cpp

```diff
@@ -75,6 +76,8 @@ void ApplyRule::AddRule(const String& sourceType, const String& targetType, cons
 	const Expression::Ptr& expression, const Expression::Ptr& filter, const String& package, const String& fkvar,
 	const String& fvvar, const Expression::Ptr& fterm, bool ignoreOnError, const DebugInfo& di, const Dictionary::Ptr& scope)
 {
+	ActivationContext::AssertOnContext();
+
 	m_Rules[sourceType].push_back(ApplyRule(targetType, name, expression, filter, package, fkvar, fvvar, fterm, ignoreOnError, di, scope));
 }
```

Comment:
Why do this here and not also within `ApplyExpression::DoEvaluate()`?

`DoEvaluate` evaluates other expressions before (indirectly) calling `AddRule`, so when doing something like `apply Service f() ...`, this would now still evaluate `f()` before telling you that you can't even use `apply` there.

Reply:
Check out this creative use of things crashing with a segfault:
```
object Host "h" {
	enable_active_checks = false
	check_command = "dummy"
	var h = this
	vars.dummy_text = () use (h) => {
		apply Service h.vars.dummy_text() {
			assign where false
		}
	}
}
```

Reply:
So you're saying with `object` instead of `apply` in my example, it would still crash? I think it would be nicer to - in both cases - detect that this is an invalid use of `apply` or `object` earlier, rather than after starting to evaluate the expression.